uitools/common.cmake

```diff
@@ -11,7 +11,44 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-set(CPPPATH ${CMAKE_CURRENT_LIST_DIR}/cpp/Esri/ArcGISRuntime/Toolkit)
+set(CPPPATH ${TOOLKITUI_DIR}/cpp/Esri/ArcGISRuntime/Toolkit)
```

Comment:
Both places where this file is included has this variable already set and this line is setting it again in the parent context. Consider doing the following instead to prevent surprises.

```
if(NOT CPPPATH)
  message(FATAL_ERROR "CPPPATH is expected to be defined")
endif()
```

Alternatively, instead of including inline, you could consider defining a function that takes out variables instead. Something like this

```
function(set_common_vars headers sources)
  set(CPPPATH ${TOOLKITUI_DIR}/cpp/Esri/ArcGISRuntime/Toolkit)
  set(${headers}
    ${CPPPATH}/AuthenticationController.h
    ${CPPPATH}/BasemapGalleryController.h
    # ...
    PARENT_SCOPE
  )
  set(${sources}
    # ...
    PARENT_SCOPE
  )
function()
```

Reply:
Thanks for pointing that out, I ended up going with your second suggestion.